docs/io.cozy.files_metadata.md

```diff
@@ -0,0 +1,391 @@
+- `accountNumber` : {string} Number of the related account
+- `bankName` : {string} Name of the related bank
+
+#### Mail
+
+- `classification` : {string} `'mail'`
+- `datetime` : {date} Equals to `issueDate`
+- `datetimeLabel` : {string} `'issueDate'`
+- `contentAuthor` : {string}
+- `categories` : {array of strings} [`'insurance'`, `'employment'`, `'health'`, `'energy'`, `'phone'`, `'isp'`, `'real_estate'`, `'tax'`]
+
+#### Report
+
+- `classification` : {string} `'mail'`
+- `datetime` : {date} Equals to `issueDate`
+- `datetimeLabel` : {string} `'issueDate'`
+- `contentAuthor` : {string}
+- `subjects` : {array of strings} [`'theft'`, `'loss'`, `'damage'`, `accident`]
+
+#### Office
+
+- `classification` : {string} `'office'`
+- `subClassification` : {string} [`'textpad'`, `'spreadsheet'`, `'slides'`]
+
+
+### Examples
+
+#### Invoices, payment statements, payment schedules
+```js
+// invoice - bouygues telecom
+'metadata': {
+  'classification': 'invoicing',
+  'datetime': '2019-05-10T23:00:00.000Z',
+  'datetimeLabel': 'issueDate',
+  'contentAuthor': 'bouyguestelecom',
+  'categories': ['phone','isp','telecom'],
+  'subClassification': 'invoice',
+  'issueDate': '2019-05-10T23:00:00.000Z',
+  'contractReference': '0645874398',
+  'invoiceNumber': 'KJF949875',
+  'isSubscription': true,
+},
+'cozyMetadata': {
+  ...
+}
+```
+
+```js
+// payment statement - ameli
+'metadata': {
+  'classification': 'invoicing',
+  'datetime': '2019-05-10T23:00:00.000Z',
+  'datetimeLabel': 'issueDate',
+  'contentAuthor': 'ameli',
+  'categories': ['insurance', 'health'],
+  'subClassification': 'payment_statement',
+  'issueDate': '2019-05-10T23:00:00.000Z',
+},
+'cozyMetadata': {
+  ...
+}
+```
+
+```js
+// payment schedule - EDF
+'metadata': {
+  'classification': 'invoicing',
+  'datetime': '2019-05-10T23:00:00.000Z',
+  'datetimeLabel': 'issueDate',
+  'contentAuthor': 'edf',
+  'categories': ['energy'],
+  'subClassification': 'paiement_schedule',
+  'iSubscription': true,
+  'issueDate': '2019-05-10T23:00:00.000Z',
+},
+'cozyMetadata': {
+  ...
+}
+```
+
+#### Taxe notices
+
+```js
+// taxe notice on income
+'metadata': {
+  'classification': 'tax_notice',
+  'datetime': '2019-05-10T23:00:00.000Z',
+  'datetimeLabel': 'issueDate',
+  'contentAuthor': 'impots.gouv',
+  'subjects': ['income'],
+  'issueDate': '2019-05-10T23:00:00.000Z',
+},
+'cozyMetadata': {
+  ...
+}
+```
+```js
+// tax return on income
+'metadata': {
+  'classification': 'tax_report',
+  'datetime': '2019-05-10T23:00:00.000Z',
+  'datetimeLabel': 'issueDate',
+  'contentAuthor': 'impots.gouv',
+  'subjects': ['income'],
+  'formReference': '2042RICI',
+  'issueDate': '2019-05-10T23:00:00.000Z',
+},
+'cozyMetadata': {
+  ...
+}
+```
+
+#### Payslips
+
+```js
+// payslip - cozycloud
+'metadata': {
+  'classification': 'payslip',
+  'datetime': '2019-05-01T23:00:00.000Z',
+  'datetimeLabel': 'startDate',
+  'contentAuthor': 'cozycloud',
+  'startDate': '2019-05-01T23:00:00.000Z',
```

Comment:
Confusing example of a datetime
Supposing the paylips is edited the 1st
the correct datetime should be: '2019-04-30T23:00:00.000Z', it will design always the first of may at 00h00 in french datetime.
See general review for solution.

Reply:
@edas what do you think about this ?

When we don't have the exact time and only de date, we usually set the time at 23:00 Zoulou. But it's not accurate, and it's confusing to use.

So we're thinking about these 2 other solutions : 
* 1. use : '2019-05-01T00:00:00.000Z'
* 2. use : '2019-05-01'

The option 2 seems the most accurate because the fact that we don't have the exact time is more explicit, but we're afraid of compatibility issues with existing code (we're talking about `metadata.datetime`)

Reply:
Always use a date ('2019-05-01') when you aim at a day. Always use a datetime ('2019-04-30T23:00:00.000Z') when you aim at a specific time in this day.

Please, never use a datetime for a date. This seems ok today because we only have metropolitain French users but we *will* have problems once we have people in multiple time zones.

If you have other places in the code where you use a datetime for a date, please list them in an issue so we can fix them ASAP.

There is no fear to have for compatibility: No code is using `startDate` today (and if we have one, let's fix it)

Reply:
Does it mean that attributes have to be either "always date" or "always datetime" ?

The compatibility issue is actually on `metadata.datetime`.
It is used for image files to specify the datetime of the actual shot.
And now we use it for pdf documents, which come mostly with dates than datetimes.

Reply:
Yep, You can't have the same attribute with sometimes a date and sometimes a datetime. Well, you could be you will have unknown behaviour when one will read a datetime from a date. Sometimes, with the wrong timezone, you may end up with the wrong date.

The real question is up to you: What behaviour will you expect?

- a doc at 00:00:00Z
- a doc without known time but from a known UTC+1 timezone
- a doc without known time but from a known UTC+1 timezone but in summer legal time (so UTC+2)
- a doc without known time and without a known timezone 
- a doc at 00:00:30 at UTC+01:00

Then:
- How do you sort them? if I am in UTC+1 timezone ? in UTC+1 with summer legal time ? in UTC+6 timezone ? in UTC-6 timezone ?
- What do you display for each doc in each case?

I let you put that in tables, then we'll discuss the technical solution (but they may be complex or require a complexe time management library) 

Until them, the rules for developpers are: 
- do not put a date in a datetime format
- do not mix a dates and datetimes in the same attribute if you want to sort them with this attribute
- always have a timezone in datetime
- do not translate timezone in the raw data (01:00:00+01:00 should not be written as 00:00:00Z)
- if you really need complex time management, add a text timezone (like Europe/Paris) aside with the date or datetime

Reply:
For administrative documents, all the dates don't have a precise time.
So, every date attribute is set as a 'date' format.
As we discussed, we will finally mix date and datetime format in datetime.